src/lightning_app/storage/drive.py

```diff
@@ -34,6 +34,12 @@ def __init__(
                 When not provided, it is automatically inferred by Lightning.
             root_folder: This is the folder from where the Drive perceives the data (e.g this acts as a mount dir).
         """
+        if id.startswith("s3://"):
```

Comment:
This is needed as per the acceptance criteria in [the ticket](https://lightningai.atlassian.net/browse/ENG-502): 

```
Passing an s3 path into a Drive will result in an error telling users to use a Mount
```

Reply:
We don't want users to be confused between drives & mounts, and instead of just saying "only protocol `lit://` is allowed, we should indicate that the users should use the `Mount` class instead